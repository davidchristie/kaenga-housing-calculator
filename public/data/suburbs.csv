Name,Average house price,Area,Reference suburb
CBD,1857308,CBD,Grey Lynn
Epsom,1521442,Central,Grey Lynn
Freemans Bay,1512500,Central,Grey Lynn
Greenlane,1120700,Central,Grey Lynn
Grey Lynn,1262523,Central,Grey Lynn
Herne Bay,1194800,Central,Grey Lynn
Mt Albert,957752,Central,Grey Lynn
Mt Eden,1359587,Central,Grey Lynn
Mt Roskill,955354,Central,Grey Lynn
One Tree Hill,1039167,Central,Grey Lynn
Parnell,1191929,Central,Grey Lynn
Ponsonby,1492000,Central,Grey Lynn
Royal Oak,1250643,Central,Grey Lynn
Sandringham,1076967,Central,Grey Lynn
Waterview,1150200,Central,Grey Lynn
Westmere,1436000,Central,Grey Lynn
Ellerslie,907132,Eastern,Grey Lynn
Glen Innes,895344,Eastern,Grey Lynn
Glendowie,1292375,Eastern,Grey Lynn
Kohimarama,1527583,Eastern,Grey Lynn
Meadowbank,1047692,Eastern,Grey Lynn
Mission Bay,1101400,Eastern,Grey Lynn
Orakei,2465333,Eastern,Grey Lynn
Point England,975860,Eastern,Grey Lynn
Remuera,1338319,Eastern,Grey Lynn
St Heliers,1569421,Eastern,Grey Lynn
St Johns,1051100,Eastern,Grey Lynn
Three Kings,865750,Eastern,Grey Lynn
Albany,758915,North Shore,New Lynn
Beach Haven,740318,North Shore,New Lynn
Birkdale,737097,North Shore,New Lynn
Birkenhead,1042865,North Shore,New Lynn
Browns Bay,854775,North Shore,New Lynn
Castor Bay,1303667,North Shore,New Lynn
Devonport,1242309,North Shore,New Lynn
Forrest Hill,1001011,North Shore,New Lynn
Glenfield,758165,North Shore,New Lynn
Greenhithe,894000,North Shore,New Lynn
Mairangi Bay,1093944,North Shore,New Lynn
Milford,1195083,North Shore,New Lynn
Murrays Bay,1091250,North Shore,New Lynn
Northcote Point,1198200,North Shore,New Lynn
Rothesay Bay,1013833,North Shore,New Lynn
Sunnynook,924609,North Shore,New Lynn
Takapuna,1287160,North Shore,New Lynn
Torbay,846790,North Shore,New Lynn
Clendon Park,496674,Southern,New Lynn
Conifer Grove,671000,Southern,New Lynn
Favona,662833,Southern,New Lynn
Flat Bush,684081,Southern,New Lynn
Goodwood Heights,721583,Southern,New Lynn
Mangere,555977,Southern,New Lynn
Mangere Bridge,1049643,Southern,New Lynn
Manukau Central,597357,Southern,New Lynn
Manurewa,563348,Southern,New Lynn
Mt Wellington,743052,Southern,New Lynn
Onehunga,955192,Southern,New Lynn
Otahuhu,597114,Southern,New Lynn
Otara,495273,Southern,New Lynn
Panmure,864083,Southern,New Lynn
Papakura,575139,Southern,New Lynn
Papatoetoe,670916,Southern,New Lynn
Takanini,654889,Southern,New Lynn
Wattle Downs,639474,Southern,New Lynn
Weymouth,520545,Southern,New Lynn
Avondale,856033,West,New Lynn
Blockhouse Bay,895878,West,New Lynn
Glen Eden,640840,West,New Lynn
Glendene,721158,West,New Lynn
Green Bay,919667,West,New Lynn
Henderson,660123,West,New Lynn
Hillsborough,975750,West,New Lynn
Hobsonville,1050222,West,New Lynn
Kelston,637731,West,New Lynn
Massey,660348,West,New Lynn
New Lynn,756818,West,New Lynn
New Windsor,922623,West,New Lynn
Ranui,605882,West,New Lynn
Royal Heights,705700,West,New Lynn
Sunnyvale,701313,West,New Lynn
Swanson,797143,West,New Lynn
Te Atatu South,728312,West,New Lynn
Te Atatu Peninsula,775924,West,New Lynn
Te Atatu South,728312,West,New Lynn
Titirangi,780575,West,New Lynn
West Harbour,886710,West,New Lynn
Whenuapai,1406500,West,New Lynn
