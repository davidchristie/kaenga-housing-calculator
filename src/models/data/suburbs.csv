City,Region,Suburb,Commute to city,3bdr hse,cost per net sqm floor area_kaenga
Auckland,Central,Avondale,21-40,863821,1282.902158
Auckland,Central,Blockhouse Bay,21-40,974880,1447.841226
Auckland,Central,Epsom,11-20,1781938,2646.441919
Auckland,Central,Freemans Bay,Walking distance,1423750,2114.479675
Auckland,Central,Greenlane,21-40,1346667,2000
Auckland,Central,Grey Lynn,11-20,1390625,2065.284142
Auckland,Central,Mt Albert,11-20,1086333,1613.365442
Auckland,Central,Mt Eden,11-20,1564823,2323.993979
Auckland,Central,Mt Roskill,21-40,925465,1374.45263
Auckland,Central,New Windsor,21-40,973700,1446.088751
Auckland,Central,One Tree Hill,21-40,1232235,1830.051527
Auckland,Central,Onehunga,21-40,1046450,1554.133279
Auckland,Central,Ponsonby,Walking distance,1513714,2248.089543
Auckland,Central,Pt Chevalier,11-20,1513714,2248.089543
Auckland,Central,Royal Oak,21-40,1210800,1798.217377
Auckland,Central,Sandringham,11-20,1375861,2043.357415
Auckland,Central,Three Kings,21-40,1162333,1726.236701
Auckland,Central,Westmere,11-20,1365400,2027.82128
Auckland,Eastern,Ellerslie,21-40,1096222,1628.052072
Auckland,Eastern,Glen Innes,21-40,1016542,1509.715468
Auckland,Eastern,Glendowie,21-40,1584722,2353.546942
Auckland,Eastern,Meadowbank,21-40,1220500,1812.623314
Auckland,Eastern,Mt Wellington,21-40,1009119,1498.691213
Auckland,Eastern,Panmure,21-40,1175900,1746.385706
Auckland,Eastern,Parnell,Walking distance,1780625,2644.49192
Auckland,Eastern,Pt England,21-40,1126500,1673.019388
Auckland,Eastern,Remuera,11-20,1447958,2150.432141
Auckland,Eastern,St Heliers,21-40,1555125,2309.591012
Auckland,Eastern,St Johns,21-40,1351250,2006.806434
Auckland,Franklin/Manukau,Pukekohe,60+,805177,860.8756549
Auckland,Franklin/Manukau,Tuakau,60+,551350,589.4900032
Auckland,Franklin/Manukau,Waiuku,60+,595300,636.4802737
Auckland,North Shore,Albany,41-60,945611,1011.02427
Auckland,North Shore,Beach Haven,21-40,879292,940.1176093
Auckland,North Shore,Birkdale,21-40,764994,817.9129691
Auckland,North Shore,Birkenhead,11-20,1069267,1143.234256
Auckland,North Shore,Browns Bay,21-40,960364,1026.797819
Auckland,North Shore,Castor Bay,21-40,1767000,1889.233401
Auckland,North Shore,Devonport,21-40,1349000,1442.317973
Auckland,North Shore,Forrest Hill,21-40,985583,1053.76136
Auckland,North Shore,Glenfield,21-40,859187,918.6218326
Auckland,North Shore,Greenhithe,21-40,1266250,1353.843687
Auckland,North Shore,Mairangi Bay,21-40,1267050,1354.699027
Auckland,North Shore,Milford,21-40,1462600,1563.776328
Auckland,North Shore,Murrays Bay,21-40,1360500,1454.613493
Auckland,North Shore,Sunnynook,21-40,1008563,1078.331017
Auckland,North Shore,Takapuna,11-20,1190471,1272.822624
Auckland,North Shore,Torbay,21-40,934636,999.2900674
Auckland,Pakuranga/Howick,Beachlands,41-60,1598333,1708.898749
Auckland,Pakuranga/Howick,Bucklands Beach,41-60,1127167,1205.139527
Auckland,Pakuranga/Howick,Dannemora,41-60,942000,1007.163477
Auckland,Pakuranga/Howick,Half Moon Bay,41-60,959972,1026.378702
Auckland,Pakuranga/Howick,Howick,41-60,894550,956.4310916
Auckland,Pakuranga/Howick,Maraetai,41-60,1166590,1247.28964
Auckland,Pakuranga/Howick,Northpark,21-40,1040513,1112.491179
Auckland,Pakuranga/Howick,Pakuranga,21-40,1097158,1173.054635
Auckland,Rodney,Arkles Bay,60+,772143,825.5565059
Auckland,Rodney,Gulf Harbour,60+,805111,860.8050893
Auckland,Rodney,Helensville,60+,814000,870.3089918
Auckland,Rodney,Manly,60+,752889,804.9705977
Auckland,Rodney,Orewa,60+,938357,1003.26847
Auckland,Rodney,Red Beach,21-40,861056,920.6201219
Auckland,Rodney,Stanmore Bay,60+,788895,843.4673367
Auckland,Rodney,Warkworth,60+,1283875,1372.687908
Auckland,South,Clendon Park,41-60,558400,597.0276916
Auckland,South,Favona,21-40,652583,697.7258634
Auckland,South,Flat Bush,21-40,765714,818.6827756
Auckland,South,Mangere,21-40,645955,690.639367
Auckland,South,Mangere Bridge,21-40,917500,980.9686732
Auckland,South,Manurewa,41-60,637567,681.6711216
Auckland,South,Otahuhu,21-40,954292,1020.305784
Auckland,South,Otara,21-40,590389,631.229552
Auckland,South,Papakura,41-60,633517,677.3409601
Auckland,South,Papatoetoe,21-40,760058,812.635518
Auckland,South,Takanini,41-60,668250,714.4766385
Auckland,South,The Gardens,41-60,925375,989.3884315
Auckland,South,Totara Heights,41-60,686600,734.096012
Auckland,South,Wattle Downs,41-60,683423,730.6992409
Auckland,South,Weymouth,41-60,676400,723.1904202
Auckland,West,Glen Eden,21-40,708549,757.5633487
Auckland,West,Glendene,21-40,750464,802.3778467
Auckland,West,Henderson,21-40,760510,813.1187854
Auckland,West,Hobsonville,21-40,811195,867.309954
Auckland,West,Kelston,21-40,727000,777.2907089
Auckland,West,Massey,21-40,735808,786.7080081
Auckland,West,New Lynn,21-40,748240,800
Auckland,West,Ranui,21-40,645854,690.5313803
Auckland,West,Royal Heights,21-40,806438,862.2238854
Auckland,West,Sunnyvale,41-60,716067,765.6014113
Auckland,West,Swanson,41-60,793364,848.2454827
Auckland,West,Te Atatu Peninsula,21-40,902905,965.3640543
Auckland,West,Te Atatu South,21-40,869571,929.7241527
Auckland,West,Titirangi,41-60,848850,907.5697637
Auckland,West,Waitakere,41-60,823750,880.7334545
Auckland,West,West Harbour,21-40,882235,943.2641933
Wellington,Central,Thorndon,Walking distance,740000,259.6491228
Wellington,Central,Pipitea,Walking distance,650000,228.0701754
Wellington,Central,Oriental Bay,Walking distance,1835000,643.8596491
Wellington,Central,Mount Victoria,Walking distance,910000,319.2982456
Wellington,Central,Te Aro,Walking distance,655000,229.8245614
Wellington,Central,Brooklyn,11-20,700000,245.6140351
Wellington,Central,Aro Valley,Walking distance,642500,225.4385965
Wellington,Central,Highbury,11-20,790000,277.1929825
Wellington,Central,Kelburn,Walking distance,950000,333.3333333
Wellington,Northern,Broadmeadows,11-20,540000,200
Wellington,Northern,Churton Park,11-20,595000,220.3703704
Wellington,Northern,Glenside,11-20,500000,185.1851852
Wellington,Northern,Grenada,21-40,545000,201.8518519
Wellington,Northern,Grenada North,21-40,380000,140.7407407
Wellington,Northern,Johnsonville,11-20,575000,212.962963
Wellington,Northern,Newlands,11-20,545000,201.8518519
Wellington,Northern,Ohariu,21-40,597500,221.2962963
Wellington,Northern,Paparangi,11-20,545000,201.8518519
Wellington,Northern,Tawa,21-40,525000,194.4444444
Wellington,Northern,Takapu Valley,21-40,650000,240.7407407
Wellington,Northern,Woodridge,11-20,535000,198.1481481
Wellington,Eastern,Hataitai,11-20,805000,298.1481481
Wellington,Eastern,Lyall Bay,21-40,730000,270.3703704
Wellington,Eastern,Kilbirnie,11-20,680000,251.8518519
Wellington,Eastern,Miramar,21-40,705000,261.1111111
Wellington,Eastern,Seatoun,21-40,1025000,379.6296296
Wellington,Eastern,Breakers Bay,21-40,875000,324.0740741
Wellington,Eastern,Karaka Bays,21-40,990000,366.6666667
Wellington,Eastern,Maupuia,21-40,652500,241.6666667
Wellington,Eastern,Melrose,21-40,720000,266.6666667
Wellington,Eastern,Moa Point,21-40,805000,298.1481481
Wellington,Eastern,Rongotai,21-40,655000,242.5925926
Wellington,Eastern,Roseneath,11-20,855000,316.6666667
Wellington,Eastern,Strathmore,21-40,655000,242.5925926
Wellington,Southern,Berhampore,21-40,650000,228.0701754
Wellington,Southern,Island Bay,21-40,740000,259.6491228
Wellington,Southern,Newtown,11-20,660000,231.5789474
Wellington,Southern,Vogeltown,11-20,665000,233.3333333
Wellington,Southern,Houghton Bay,21-40,682500,239.4736842
Wellington,Southern,Kingston,21-40,645000,226.3157895
Wellington,Southern,Mornington,21-40,665000,233.3333333
Wellington,Southern,Mount Cook,Walking distance,660000,231.5789474
Wellington,Southern,Owhiro Bay,21-40,610000,214.0350877
Wellington,Southern,Southgae,21-40,695000,243.8596491
Wellington,Western,Karori,21-40,665000,246.2962963
Wellington,Western,Northland,11-20,715000,264.8148148
Wellington,Western,Crofton Downs,21-40,630000,233.3333333
Wellington,Western,Kaiwharawhara,11-20,675000,250
Wellington,Western,Khandallah,11-20,760000,281.4814815
Wellington,Western,Ngaio,11-20,690000,255.5555556
Wellington,Western,Ngauranga,11-20,660000,244.4444444
Wellington,Western,Makara,21-40,535000,198.1481481
Wellington,Western,Makara Beach,21-40,472500,175
Wellington,Western,Wadestown,11-20,800000,296.2962963
Wellington,Western,Wilton,11-20,675000,250
Wellington,Lower Hutt,Petone,21-40,660000,244.4444444
Wellington,Lower Hutt,Melling,21-40,580000,214.8148148
Wellington,Lower Hutt,Maungaraki,21-40,565000,209.2592593
Wellington,Lower Hutt,Normandale,21-40,550000,203.7037037
Wellington,Lower Hutt,Kelson,21-40,485000,179.6296296
Wellington,Lower Hutt,Belmont,21-40,495000,183.3333333
Wellington,Lower Hutt,Alicetown,21-40,620000,229.6296296
Wellington,Lower Hutt,Moera,21-40,495000,183.3333333
Wellington,Lower Hutt,Woburn,21-40,715000,264.8148148
Wellington,Lower Hutt,Waiwhetu,21-40,520000,192.5925926
Wellington,Lower Hutt,Waterloo,21-40,585000,216.6666667
Wellington,Lower Hutt,Taita,21-40,350000,129.6296296
Wellington,Lower Hutt,Avalon,21-40,500000,185.1851852
Wellington,Lower Hutt,Stokes Valley,21-40,370000,137.037037
Wellington,Lower Hutt,Waomiuomata,21-40,300000,111.1111111
Wellington,Lower Hutt,Eastbourne,41-60,730000,270.3703704
Wellington,Lower Hutt,Fairfield,21-40,535000,198.1481481
Wellington,Lower Hutt,Epuni,21-40,565000,209.2592593
Wellington,Lower Hutt,Gracefield,21-40,445000,164.8148148
Wellington,Lower Hutt,Point Howard,21-40,695000,257.4074074
Wellington,Lower Hutt,Lowry Bay,21-40,845000,312.962963
Wellington,Lower Hutt,York Bay,21-40,665000,246.2962963
Wellington,Lower Hutt,Mahina Bay,21-40,737500,273.1481481
Wellington,Lower Hutt,Sunshine Bay,41-60,757500,280.5555556
Wellington,Lower Hutt,Days Bay,41-60,720000,266.6666667
Wellington,Porirua,Aotea,21-40,750000,277.7777778
Wellington,Porirua,Ascot Park,21-40,375000,138.8888889
Wellington,Porirua,Camborne,21-40,555000,205.5555556
Wellington,Porirua,Cannons Creek,21-40,265000,98.14814815
Wellington,Porirua,Elsdon,21-40,340000,125.9259259
Wellington,Porirua,Hongoeka,21-40,540000,200
Wellington,Porirua,Kenepuru,21-40,695000,257.4074074
Wellington,Porirua,Papkowhai,21-40,560000,207.4074074
Wellington,Porirua,Paremata,21-40,550000,203.7037037
Wellington,Porirua,Pauatahanui,21-40,557500,206.4814815
Wellington,Porirua,Plimmerton,21-40,662000,245.1851852
Wellington,Porirua,Pukerua Bay,21-40,540000,200
Wellington,Porirua,Ranui,21-40,330000,122.2222222
Wellington,Porirua,Takapuwahia,21-40,335000,124.0740741
Wellington,Porirua,Titahi Bay,21-40,420000,155.5555556
Wellington,Porirua,Waitangirua,21-40,255000,94.44444444
Wellington,Porirua,Whitby,21-40,530000,196.2962963
Christchurch,Northern,Mairehau,Walking distance,420000,189.0909091
Christchurch,Northern,Shirley,11-20,390000,189.0909091
Christchurch,Northern,Dallington,11-20,395000,189.0909091
Christchurch,Central,Richmond,Walking distance,385000,175.5844156
Christchurch,Eastern,Avonside,11-20,350000,177.8354978
Christchurch,Central,Linwood,11-20,350000,173.3333333
Christchurch,Central,Phillipstown,Walking distance,320000,157.5757576
Christchurch,Eastern,Woolston,11-20,345000,157.5757576
Christchurch,Southern,Opawa,11-20,490000,144.0692641
Christchurch,Central,Waltham,Walking distance,350000,155.3246753
Christchurch,Soothern,St Martins,11-20,480000,220.6060606
Christchurch,Southern,Beckenham,11-20,550000,157.5757576
Christchurch,Central,Sydenham,Walking distance,475000,216.1038961
Christchurch,Southern,Somerfield,11-20,475000,247.6190476
Christchurch,Southern,Spreydon,11-20,430000,213.8528139
Christchurch,Central,Addington,Walking distance,395000,213.8528139
Christchurch,Central,Riccarton,11-20,510000,193.5930736
Christchurch,Western,Ilam,11-20,620000,177.8354978
Christchurch,Western,Upper Riccarton,11-20,485000,229.6103896
Christchurch,Western,Burnside,11-20,510000,279.1341991
Christchurch,Western,Fendalton,11-20,790000,218.3549784
Christchurch,Northern,Bryndwr,11-20,485000,229.6103896
Christchurch,Northern,Strowan,11-20,655000,355.6709957
Christchurch,Northern,Merival,11-20,815000,218.3549784
Christchurch,Northern,Papnui,11-20,485000,294.8917749
Christchurch,Northern,St Albans,11-20,530000,366.9264069
Christchurch,Northern,Edgeware,11-20,420000,218.3549784