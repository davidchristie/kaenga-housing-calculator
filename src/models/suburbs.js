import { csv } from 'd3-request'

export const readSuburbsData = callback => {
  csv('/data/suburbs.csv', (error, data) => {
    if (error) callback(error)
    const suburbs = data.map(row => {
      return {
        area: row['Area'],
        averageHousePrice: row['Average house price'],
        name: row['Name'],
        referenceSuburb: row['Reference suburb']
      }
    })
    callback(null, suburbs)
  })
}
