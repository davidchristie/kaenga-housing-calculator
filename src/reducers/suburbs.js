export default (state = [], action) => {
  switch (action.type) {
    case 'LOAD_SUBURBS':
      return action.suburbs

    default:
      return state
  }
}
