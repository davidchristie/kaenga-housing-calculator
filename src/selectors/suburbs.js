export const getSuburbs = state => {
  return state.suburbs
}
