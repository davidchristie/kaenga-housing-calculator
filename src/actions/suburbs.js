import { readSuburbsData } from '../models/suburbs'

export const loadSuburbs = () => {
  return dispatch => {
    readSuburbsData((error, suburbs) => {
      if (error) throw error
      dispatch({
        suburbs,
        type: 'LOAD_SUBURBS'
      })
    })
  }
}
