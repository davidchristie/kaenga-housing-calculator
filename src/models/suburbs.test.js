import { readSuburbsData } from './suburbs'

it('matches snapshot', done => {
  readSuburbsData(data => {
    expect(data).toMatchSnapshot()
    done()
  })
})
