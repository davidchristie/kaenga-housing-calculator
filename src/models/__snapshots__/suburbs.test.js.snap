exports[`test matches snapshot 1`] = `null`;
